__tests__/core/type-system.test.ts

```diff
@@ -22,6 +22,10 @@ import {
 } from "../../src"
 import { $nonEmptyObject } from "../../src/internal"
 
+const assertTypesEqual = <T, U>(t: T, u: Exact<T, U> extends never ? T : U): [T, U] =>
```

Comment:
`Exact<T, U> extends never ? T : U` when it's not exact we make the type of the second arg be the same type as the first. That way we lean on TS giving us an error about the specifics:

![vscode screenshot with a clarifying type error](https://github.com/mobxjs/mobx-state-tree/assets/833576/d673169e-726a-4a04-bc49-b786e8175ed4)

Reply:
Phenomenal. This is what I really needed when I was first trying to debug the old `spec.ts` failures. Thank you!